### PR TITLE
[#67] Logs → Preserve text selection during refresh

### LINKED ISSUE #67: Log view wipes text selection
The log view polls every 2 seconds and rebuilds the entire DOM list each time. Any text you've selected in the pane is destroyed on every poll, so you can't reliably select and copy a log line.

This matters because the log view exists so users can copy log lines into a support request. The core action of the feature is the one that breaks.

### PR DESCRIPTION
**Pull request by terminalchai**
Resolves [#67](https://github.com/8thpark/geode/issues/67)

## Problem

The log pane rebuilds its DOM when new entries arrive. If a user is selecting a log line to copy into a support request, that redraw destroys the selection before it can be copied. The polling described in the original issue has since been replaced by live log events, but an event arriving during selection still causes the same failure.

## Changes

- Detect a non-collapsed browser selection that intersects the log pane
- Defer redraws while that selection remains active
- Refresh from the persisted sink as soon as the selection clears, so deferred entries are not lost
- Add focused tests for collapsed, empty, outside, and intersecting selections

## Validation

- `npm run lint`
- `npm run build`
- `npm test` (279 tests)
- `npm run check-versions`
- `npm run audit`

`npm run test:integration` was not run locally because this is an isolated log-view change with no storage or sync behavior; the repository CI will still run that Docker-backed suite.

<h3>Greptile Summary</h3>

This PR preserves log text selection by deferring log-pane redraws while an intersecting browser selection is active and refreshing from the persisted sink when it clears.
- Adds selection-overlap and render-transition helpers.
- Integrates deferred rendering with the log view's existing refresh coalescing.
- Adds focused unit tests for selection states and transition decisions.

<h3>Confidence Score: 5/5</h3>

The PR appears safe to merge because no blocking failure remains.

The reply stating “Addressed in 58dfd77” does not fully resolve the previously reported coverage gap: the new test drives `nextRender` directly, while the actual `GeodeLogView` listener, deferred state, and catch-up refresh remain unexercised.

**Files Needing Attention:** src/log/selection.test.ts, src/log/view.ts

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/log/selection.ts | Adds pure helpers for detecting selection overlap and deciding deferred render transitions. |
| src/log/view.ts | Integrates selection-aware deferral and catch-up refreshes with the existing log-view refresh loop. |
| src/log/selection.test.ts | Covers selection-overlap cases and the deferred render-state transition sequence. |

<sub>Reviews (2): Last reviewed commit: ["Logs: cover deferred refresh lifecycle"](https://github.com/8thpark/geode/commit/58dfd778c674f32e189eb718b56fc99152de062d) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49512309)</sub>

**Context used:**

- Knowledge Base — [Log module](https://app.greptile.com/8thpark/-/custom-context/knowledge-base/8thpark/geode/-/docs/log-module.md)

### CODE REVIEW

**Comment by revett** [comment]
@greptileai Review PR

File: src/log/selection.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const selection = fakeSelection(false, 1, false);
+
+  assert.equal(selectionOverlaps(container, selection), false);
+});
+
+test("selectionOverlaps: selection intersecting the log defers rendering", () => {
+  const selection = fakeSelection(false, 1, true);
+
+  assert.equal(selectionOverlaps(container, selection), true);
+});
```
[the remark this answers is not in the document]
Comment (terminalchai) [reply src/log/selection.test.ts:45]: Addressed in 58dfd77. The view now delegates deferred-render state to the pure `nextRender` transition, and the lifecycle test covers deferral while selected, continued waiting, the single catch-up refresh when selection clears, and no duplicate refresh afterward.

**Comment by revett** [comment]
@greptileai Review PR

**Review by revett — APPROVED** [review]
@terminalchai Thanks for another contribution! 🍪
